lib/src/Base/Stat/openturns/HMatrixImplementation.hxx

```diff
@@ -134,8 +135,10 @@ public:
   UnsignedInteger getNbColumns() const;
 
   void assemble(const HMatrixRealAssemblyFunction& f, char symmetry);
-  void assemble(const HMatrixTensorRealAssemblyFunction& f, char symmetry);
-  void factorize(const String& method);
+  void assemble(const HMatrixRealAssemblyFunction &f, const HMatrixParameters & parameters, char symmetry);
+  void assemble(const HMatrixTensorRealAssemblyFunction &f, char symmetry);
+  void assemble(const HMatrixTensorRealAssemblyFunction &f, const HMatrixParameters &parameters, char symmetry);
```

Comment:
Why not define the 3-argument function as
```cpp
void assemble(const HMatrixTensorRealAssemblyFunction &f, char symmetry, const HMatrixParameters &parameters);
```
That way, you could view `parameters` as an optional argument in the doc.

Reply:
Because if we follow the coding rules, we start with complex structures and after more basic objects (int/char/...)